modules/cart/commerce_cart.services.yml

```diff
@@ -9,7 +9,7 @@ services:
 
   commerce_cart.cart_manager:
     class: Drupal\commerce_cart\CartManager
-    arguments: ['@entity_type.manager', '@commerce_cart.order_item_matcher', '@event_dispatcher']
+    arguments: ['@entity_type.manager', '@commerce_cart.order_item_matcher', '@event_dispatcher', '@commerce_order.order_refresh']
```

Comment:
Discussed with @bojanz, we decided that orders just need a method to flag them as "dirty" so that on save they refresh themselves.